pdr_backend/lake/etl.py

```diff
@@ -56,27 +77,64 @@ def _drop_temp_sql_tables(self):
         db = DuckDBDataStore(self.ppss.lake_ss.lake_dir)
         for dataclass in _ETL_REGISTERED_LAKE_TABLES:
             table_name = dataclass.get_lake_table_name()
-            db.drop_view(ETLTable(table_name).fullname)
-            db.drop_table(TempTable(table_name).fullname)
+            db.drop_table(TempTable(table_name).table_name)
+            db.drop_table(NewEventsTable(table_name).table_name)
+            db.drop_table(UpdateEventsTable(table_name).table_name)
+            db.drop_table(TempUpdateTable(table_name).table_name)
 
-    def _move_from_temp_tables_to_live(self):
+    def _do_bronze_swap_to_prod_atomic(self):
         """
         @description
-            Move the records from our bronze temporary tables to live, in-production tables
+            Merge the bronze ETL tables against prod database
+            This needs to become a single transaction per table
+            Such that it remains atomic, and is able to resume if it fails
         """
-
         db = DuckDBDataStore(self.ppss.lake_ss.lake_dir)
-        for dataclass in _ETL_REGISTERED_LAKE_TABLES:
-            table_name = dataclass.get_lake_table_name()
-            logger.info("move table %s to live", table_name)
-            temp_table = TempTable(table_name)
-            permanent_table = NamedTable(table_name)
 
-            if db.table_exists(temp_table.fullname):
-                db.move_table_data(temp_table, permanent_table)
+        bronze_tables = [BronzePrediction]
+        for table in bronze_tables:
+            prod_table = Table.from_dataclass(table)
```

Comment:
I think some of the contents of this for can be on the Bronze Table class itself. The object should know how to move itself into the next step.

Reply:
You can add a mixin on all Bronze tables that abstracts this functionality.

Reply:
tagging @trentmc here as I'm concerned that this could lead to more complex problems...

No. **Please separate the concerns of the ETL pipeline from the Table.**

Read the readme...
Tables know nothing about ETL.
This is done very intentionally...

**The ETL may eventually be executed in a SQL cluster, or a distributed MPP pipeline. None of this should exist inside of "BronzeTable". Bronze table is just a wrapper for table schema/validation/etc...**

Please be more considerate of **LoB** - Locality of Behavior... The ETL is responsible for bronze/silver/gold/ ETL processes/swap/etc... Tables should know nothing about this...

![image](https://github.com/user-attachments/assets/1a46c1dd-5739-4145-902f-8da11c6c7446)